src/components/TaskList/TaskList.jsx

```diff
@@ -9,68 +9,103 @@ const TaskVisibility = require('../../models/TaskVisibility')
 class TaskList extends React.Component {
   constructor(props) {
     super(props)
-    this.state = { selectedIndex: null }
-    this.onKeyUp = this.onKeyUp.bind(this)
-    this.onKeyDown = this.onKeyDown.bind(this)
+    this.state = { focusedIndex: 0, showHidden: false }
   }
 
   componentDidMount() {
-    document.addEventListener('keyup', this.onKeyUp)
-    document.addEventListener('keydown', this.onKeyDown)
+    this.keyDownEventListener = event => this.onKeyDown(event)
```

Comment:
I switched to this because it felt weird to reassign an instance class from within the instance. I'd prefer if we could just do 

```js
 document.addEventListener('keydown', this.onKeyDown)

// and

 document.removeEventListener('keydown', this.onKeyDown)
```

But Javascript is evil and likes to make things difficult for us :sad: